### PR TITLE
【Hackathon No.7】为 Paddle 新增 apply API -part

### PR DESCRIPTION
**Pull request by yangguohao**
### PR types
New features

### PR changes
APIs

### Description
add tensor.apply for Paddle 
RFC:https://github.com/PaddlePaddle/community/pull/761

### CODE REVIEW

**Comment by luotao1** [comment]
image
如果这部分单测测到了，可以补充下本地截图

**Comment by yangguohao** [comment]
> image 如果这部分单测测到了，可以补充下本地截图

添加了单测

**Comment by yangguohao** [comment]
@luotao1 @jeff41404

File: python/paddle/base/dygraph/tensor_patch_methods.py [hunk 1]
```diff
@@ -383,6 +383,62 @@ def gradient(self):
...
+            .. code-block:: python
+
+                >>> import paddle
+
+                >>> x = paddle.to_tensor(5.)
+                >>> f = lambda x: 3*x+2
+                >>> x.apply_(f)
+                >>> print(x)
+                Tensor(shape=[], dtype=float32, place=Place(cpu), stop_gradient=True,
+                 17.)
```
Comment (jeff41404) [inline python/paddle/base/dygraph/tensor_patch_methods.py]: can the shape of the input data in example code be larger, such as [3, 3]

Comment (jeff41404) [reply python/paddle/base/dygraph/tensor_patch_methods.py]: and use tensor in GPU

File: test/legacy_test/test_apply.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import unittest
+
+import numpy as np
+
+import paddle
+
+
+class TestTensorApplyAPI(unittest.TestCase):
+    def setUp(self):
+        self.x = paddle.to_tensor([1, 2, 3, 4, 5], stop_gradient=True)
```
Comment (jeff41404) [inline test/legacy_test/test_apply.py:24]: we need test input tensor in both CPU and GPU

File: python/paddle/base/dygraph/tensor_patch_methods.py (hunk 1 above)
Comment (jeff41404) [reply python/paddle/base/dygraph/tensor_patch_methods.py]: are we support other data types? eg `float16`, `bfloat16` and `float64`, should add more case in example code

Comment (jeff41404) [reply python/paddle/base/dygraph/tensor_patch_methods.py]: same issue in example code in `def apply` below

File: test/legacy_test/test_apply.py (hunk 2 above)
Comment (jeff41404) [reply test/legacy_test/test_apply.py:24]: are we support other data types? eg float16, bfloat16 and float64. if yes, should add more case in here

File: python/paddle/base/dygraph/tensor_patch_methods.py (hunk 1 above)
Comment (yangguohao) [reply python/paddle/base/dygraph/tensor_patch_methods.py]: done add the related example code

File: test/legacy_test/test_apply.py (hunk 2 above)
Comment (yangguohao) [reply test/legacy_test/test_apply.py:24]: Done add the test case

**Review by jeff41404 — DISMISSED** [review]
LGTM

**Comment by luotao1** [comment]
请提交对应的中文文档

**Review by sunzhongkai588 — COMMENTED** [review]
代码示例没过，检查一下 CI-Static-check 的报错log

**Comment by yangguohao** [comment]
> 代码示例没过，检查一下 CI-Static-check 的报错log

我发现好像 CI 里 paddle.seed 固定种子数后输出的随机值和我电脑本地的输出不一样，打印的时候会出现问题。

**Review by sunzhongkai588 — DISMISSED** [review]
LGTM

**Comment by luotao1** [comment]
@yangguohao 请解决下冲突

**Comment by yangguohao** [comment]
> @yangguohao 请解决下冲突

done 等 CI 完应该就 ok 了

**Review by luotao1 — APPROVED** [review]

**Pull request merged**